server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -373,7 +427,7 @@ synchronized void updateRemoteCluster(
             remote = new RemoteClusterConnection(finalSettings, clusterAlias, transportService, remoteClusterCredentialsManager);
             remoteClusters.put(clusterAlias, remote);
             remote.ensureConnected(listener.map(ignored -> RemoteClusterConnectionStatus.CONNECTED));
-        } else if (remote.shouldRebuildConnection(newSettings)) {
+        } else if (forceRebuild || remote.shouldRebuildConnection(newSettings)) {
```

Comment:
nit: Since we know that this will always be true when we reload credentials, it might make sense to break this in to its own method to avoid the boolean in the method signature.

Reply:
agreed that the flag is not ideal but having stared at the method for a bit I don't see a good way to refactor it without basically duplicating it or using some sort of boolean flag in the end -- did you have a specific tweak in mind? partly, I wanted to avoid touching too much of the connection rebuilding code since it's fairly complex and outside of the security team's scope.

Reply:
Makes sense! It's fairly complex so I think it's out of scope to refactor any of that. 👍